### ISSUE TITLE
Allow `/admin` endpoints to be configurable

### ISSUE THREAD

**Issue by phlax** [envoyproxy/envoy#32051]
It would be great if it were possible to configure admin endpoints according to a (matched?) allowlist and/or denylist

This would make exposing the admin endpoint much more secure if you dont want or need all of the admin functionality

One usage i can think of immediately would be for adding a healthcheck to released docker images (cf #31619)

(This is a follow up from a related Slack conversation)

**Comment by phlax** [comment]
i think something like this could be partially acheived by proxying the admin endpoint but it would be better/more secure to be able to prevent any access

**Comment by phlax** [comment]
cc @jmarantz @abraunton

**Comment by jmarantz** [comment]
I wanted to point out also, that anyone using the C++ API (link your own main.cc replacement) can use the C++ adminRequest API to expose only the desired endpoints.

I agree it would be convenient to let people specify admin endpoint subsets as config, and should not be difficult to implement that.

**Comment by phlax** [comment]
this came up a couple of times at kubecon - would be great to get a resolution on it

**Comment by willianpaixao** [comment]
@phlax I believe this issue can be closed, right?

**Comment by phlax** [comment]
i _think_ it has not been resolved - afaict linked PR was never landed - do you think it was fixed elsewhere?

**Comment by willianpaixao** [comment]
The feature isn't covered by https://github.com/envoyproxy/envoy/pull/42367?

**Comment by phlax** [comment]
you're absolutely correct! i forgot that had landed - thanks for the headsup - i probably should update some docs/examples
